rust/gitxet/tests/integration_tests/test_integration_scenarios.sh

```diff
@@ -0,0 +1,516 @@
+git xet install 
+
+
+base_dir="`pwd`"
+mkdir config_files/
+
+
+new_bare_repo() { 
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+
+    rm -rf "$repo_dir"
+    mkdir "$repo_dir"
+    pushd "$repo_dir"
+
+    git init --bare
+
+    popd
+}
+
+# Use the local versions of the 
+new_repo() {
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+
+    rm -rf "$repo_dir"
+    mkdir "$repo_dir"
+    pushd "$repo_dir"
+
+    git init
+
+
+    popd
+}
+
+create_file_in_repo() {
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+    local file_name="$repo_dir/$2"
+
+    pushd "$repo_dir"
+    create_data_file "$file_name" 100000
+    git add "$file_name"
+    git commit -m "Added file $file_name"
+    popd
+}
+
+init_repo_as_xet () { 
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+    pushd "$repo_dir"
+
+    config_file_name="$base_dir/config_files/xet_config_file_${name}.toml"
+    
+    echo "[upstream]
+    url = \"${repo_dir}\"
+    " > "${config_file_name}" 
+
+    git xet init -m 2 --force --explicit --write-repo-salt --write-gitattributes --xet-config-file="${config_file_name}"
+
+    popd
+}
+
+intermediate_counter=0
+
+new_tmp_bare_repo () { 
+    local tmp_repo="tmp_repo_${intermediate_counter}"
+    ((intermediate_counter++))
+    >&2 new_bare_repo ${tmp_repo}
+    echo ${tmp_repo}
+}
+
+new_tmp_repo () { 
+    local tmp_repo="tmp_repo_${intermediate_counter}"
+    ((intermediate_counter++))
+    >&2 new_repo ${tmp_repo}
+    echo ${tmp_repo}
+}
+
+
+name_conuter=0
+new_name() { 
+    base=$1
+    ((name_counter++))
+    echo "base_${name_counter}"
+}
+
+
+do_push() { 
+    local repo_1="$1"
+    local repo_dir_1="$base_dir/$repo_1"
+
+    local repo_2="$2"
+    local repo_dir_2="$base_dir/$repo_2"
+
+    pushd "$repo_dir_1"
+    git push --force "$repo_dir_2" 
+    popd
+}
+
+simulate_pr_merge() { 
+    local repo_1="$1"
+    local repo_dir_1="$base_dir/$repo_1"
+
+    local repo_2="$2"
+    local repo_dir_2="$base_dir/$repo_2"
+
+    local branch=${3:-main}
+    local remote_branch=${4:-main}
+
+    # Create a temporary repo from which to do the merge. 
+    tmp_repo=$(new_tmp_repo)
+    pushd "$tmp_repo"
+
+    # Disable git from running in the intermediate bit
+    git config --local filter.xet.process ""
+    git config --local filter.xet.required false
+
+    git remote add 1 "$repo_dir_1"
+    git remote add 2 "$repo_dir_2"
+
+    git fetch 1
+    git fetch 2
+
+    git reset --hard 1/$branch
+    git merge --no-edit 2/$remote_branch
+    git push 2
+    popd
+}
+
+
+# Create a new bare repo that simulates a fork of the original repo.
+simulate_fork() {
+    local repo_1="$1"
+    local repo_dir_1="$base_dir/$repo_1"
+
+    local repo_2="$2"
+    local repo_dir_2="$base_dir/$repo_2"
+
+    # Create a temporary new bare repo to push things to it.
+    local tmp_repo=$(new_tmp_bare_repo)
+    local tmp_repo_dir="$base_dir/$tmp_repo"
+
+    do_push $repo_1 $tmp_repo 
```

Comment:
I'm interested in why it's necessary to push to a tmp repo first? (directly pushing repo_1 to repo_2 fails the test)

Reply:
I want to ensure that none of the notes make it through the fork, and it seemed the easiest way to do this is to first push to a temporary bare repo, which wouldn't ever run the filter process or set up the fetch config, and then push from there to the new forked repo.  That second push would be guaranteed to not push anything but the actual branch content. .